shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/response/GlobalResponse.kt

```diff
@@ -68,4 +68,21 @@ data class GlobalResponse(
         } else {
             null
         }
+
+    fun getRoutesFor(stopId: String): List<Route> {
+        return routePatterns.values.mapNotNull {
```

Comment:
Wouldn't using `patternIdsByStop` be more direct here?

Reply:
I tried this and I think I'm either overlooking or missing something because it only works for bus stops. Here's the code I was trying:
```
        return patternIdsByStop[stopId]?.mapNotNull { patternId ->
            val routePattern = routePatterns[patternId] ?: return@mapNotNull null
            val tripId = routePattern.representativeTripId
            val trip = trips[tripId] ?: return@mapNotNull null
            val stop = stops[stopId] ?: return@mapNotNull null
            if (
                trip.stopIds?.contains(stopId) == true ||
                    trip.stopIds?.any(stop.childStopIds::contains) == true &&
                        routePattern.typicality == RoutePattern.Typicality.Typical
            ) {
                return@mapNotNull routes[trip.routeId]
            } else {
                return@mapNotNull null
            }
        } ?: emptyList()
```
Does that seem right?

Reply:
I'd expect something like
```kotlin
val patternIds =
    if (stop.childStopIds.isEmpty()) {
        patternIdsByStop[stopId].orEmpty()
    } else {
        stop.childStopIds.flatMap { patternIdsByStop[it] }
    }
return patternIds.mapNotNull { routePatterns[it]?.routeId }.distinct().mapNotNull { routes[it] }
```
since it's the child stops that are keys in `patternIdsByStop` and the values there are already known to visit that stop.

Reply:
Ah right, I understand now. Thank you!